### DISCUSSION TITLE
Only some LSPs support server configuration schema?

### DISCUSSION THREAD

**Discussion by nyonson** [williamboman/nvim-lsp-installer#575] [Q&A] [answered]
Some of my installed servers show a "Server configuration schema" which seems like a nice way to discover how to configure a particular server. However, not seeing it under `gopls`. Is it only supported on some servers?

**Comment by williamboman** [comment, marked as answer]
Hello! Yes, that's the situation currently. Most schemas originates from its corresponding VSCode extension, where it can be found in the `$.contributes.configuration` path of the extension's package.json manifest, for example [this one](https://raw.githubusercontent.com/elixir-lsp/vscode-elixir-ls/master/package.json). This is just a regular [JSONSchema](https://json-schema.org/) structure. Unfortunately, these schemas have a tendency to also incorporate settings that only apply to the VSCode client extension itself (and not the underlying, editor-agnostic, LSP server implementation). Some servers, for example [pylsp](https://raw.githubusercontent.com/python-lsp/python-lsp-server/develop/pylsp/config/schema.json), will provide a standalone schema definition, which is really nice as it's a fully editor-agnostic schema.

These schemas are currently mapped in [this gist](https://gist.github.com/williamboman/a01c3ce1884d4b57cc93422e7eae7702), and are used in [this script](https://github.com/williamboman/nvim-lsp-installer/blob/0dd741a5dd0afb5a52488703e89758af4180dcb5/scripts/autogen_metadata.lua#L157-L185) to generate it into version controlled Lua tables (which then is used by the `:LspInstallInfo` UI). I'm sure there are servers missing in the gist, which I'd be happy to add. Long-term it'd however be really nice if LSP servers were to separate VSCode client extension from the LSP server implementation a bit more!
